storage/src/main/java/tech/pegasys/teku/storage/client/RecentChainData.java

```diff
@@ -300,7 +300,11 @@ private void updateChainHead(
         reorgCounter.inc();
         optionalReorgContext =
             ReorgContext.of(
-                previousChainHead.getRoot(), previousChainHead.getStateRoot(), commonAncestorSlot);
+                previousChainHead.getRoot(),
+                previousChainHead.getSlot(),
+                previousChainHead.getStateRoot(),
+                commonAncestorSlot,
+                getBlockRootBySlot(commonAncestorSlot).orElse(Bytes32.ZERO));
```

Comment:
I think we should be modifying `previousChainHead.findCommonAncestor` to return both the slot and root rather than having to find the common ancestor root a second time here.  `findCommonAncestor` already has to know the root because that's how it checks if the ancestor is common between the forks.  It could just return `SlotAndBlockRoot` rather than just `UInt64`.